sunbird_mcp/src/api/sandbox_search/api.py

```
@@ -0,0 +1,237 @@
+                metadata={
+                    "api": self.config.api_name,
+                    "version": response.get("ver", "1.0"),
+                    "total": count,
+                    "responseCode": response.get("responseCode"),
+                    "ts": response.get("ts"),
+                    "params": response.get("params", {})
+                }
+            )
+        except Exception as e:
+            logger.error("Error processing sandbox search response: %s", str(e), exc_info=True)
+            raise SunbirdAPIError("Failed to process sandbox search results")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)